test/JDBC/expected/BABEL-4046.out

```diff
@@ -0,0 +1,197 @@
+	[DEPTNO] [int] NOT NULL,
+PRIMARY KEY CLUSTERED 
+(
+	[EMPNO] ASC
+) ON [PRIMARY])
+GO
+
+CREATE TABLE [dbo].[t4](
+	[DEPTNO] [int] NOT NULL,
+	[DNAME] [nvarchar](14) NULL,
+	[LOC] [nvarchar](13) NULL,
+PRIMARY KEY CLUSTERED 
+(
+	[DEPTNO] ASC
+) ON [PRIMARY])
+GO
+
+INSERT [dbo].[t4] ([DEPTNO], [DNAME], [LOC]) VALUES (10, N'ACCOUNTING', N'NEW YORK')
+INSERT [dbo].[t4] ([DEPTNO], [DNAME], [LOC]) VALUES (20, N'RESEARCH', N'DALLAS')
+INSERT [dbo].[t4] ([DEPTNO], [DNAME], [LOC]) VALUES (30, N'SALES', N'CHICAGO')
+INSERT [dbo].[t4] ([DEPTNO], [DNAME], [LOC]) VALUES (40, N'SECURITY', N'BOSTON')
+INSERT [dbo].[t4] ([DEPTNO], [DNAME], [LOC]) VALUES (50, N'LEGAL', N'AUSTIN')
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7369, N'SMITH', N'BOSTON', 20)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7499, N'ALLEN', N'CHICAGO', 30)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7521, N'WARD', N'CHICAGO', 30)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7566, N'JONES', N'AUSTIN', 20)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7654, N'MARTIN', N'AUSTIN', 30)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7698, N'BLAKE', N'BOSTON', 30)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7782, N'CLARK', N'NEW YORK', 10)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7788, N'SCOTT', N'NEW YORK', 20)
+INSERT [dbo].[t3] ([EMPNO], [ENAME], [BASELOC], [DEPTNO]) VALUES (7839, N'KING', N'AUSTIN', 100)
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+
+;with EMP_T AS (
+                select empno,
+				ename,
+				CASE			            
+			            WHEN baseloc LIKE 'AUS%' THEN REPLACE(baseloc,'AUSTIN','A')
+					    WHEN baseloc LIKE 'CHI%' THEN REPLACE(baseloc,'CHICAGO','C')
+						WHEN baseloc LIKE 'BOS%' THEN REPLACE(baseloc,'BOSTON','B')
+                ELSE
+		                baseloc
+                END AS baseloc,
+				deptno
+				from t3)
+				select
+				DM.empno,
+				DM.ename,
+				DM.baseloc
+				from EMP_T DM
+				INNER JOIN t4 SR ON DM.baseloc = SR.loc AND DM.deptno = SR.deptno
+GO
+~~START~~
+int#!#nvarchar#!#text
+7782#!#CLARK#!#NEW YORK
+~~END~~
+
+
+;with EMP_T AS ( select empno, ename, CASE WHEN baseloc LIKE 'AUS%' THEN REPLACE(baseloc,'AUSTIN','A') ELSE baseloc END AS baseloc, deptno from t3)
+	select DM.empno, DM.ename, DM.baseloc from EMP_T DM where DM.baseloc in (select baseloc from t4) 
+GO
+~~START~~
+int#!#nvarchar#!#text
+7369#!#SMITH#!#BOSTON
+7499#!#ALLEN#!#CHICAGO
+7521#!#WARD#!#CHICAGO
+7566#!#JONES#!#A
+7654#!#MARTIN#!#A
+7698#!#BLAKE#!#BOSTON
+7782#!#CLARK#!#NEW YORK
+7788#!#SCOTT#!#NEW YORK
+7839#!#KING#!#A
```

Comment:
It could be better to have `order by` clause to prevent future flakeness.

Reply:
OK